### PR TITLE
[opt](Nereids) outer join with is null stats estimation enhancement

### PR DESCRIPTION
**Pull request by xzj7019**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

**Comment by xzj7019** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java [hunk 1]
```diff
@@ -404,4 +404,15 @@ public static boolean checkReorderPrecondition(LogicalJoin<?, ?> top, LogicalJoi
         markSlots.retainAll(bottom.getOutputSet());
         return markSlots.isEmpty();
     }
+
+    public static boolean isSemiOrAntiJoin(JoinType joinType) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java:408]: duplicates to JoinType.isSemiOrAntiJoin()

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java [hunk 2]
```diff
@@ -404,4 +404,15 @@ public static boolean checkReorderPrecondition(LogicalJoin<?, ?> top, LogicalJoi
...
+        return joinType == JoinType.LEFT_SEMI_JOIN || joinType == JoinType.RIGHT_SEMI_JOIN
+                || joinType == JoinType.LEFT_ANTI_JOIN || joinType == JoinType.RIGHT_ANTI_JOIN;
+    }
+
+    public static boolean isOuterJoin(JoinType joinType) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java:413]: duplicates to JoinType.isOuterJoin()

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java (hunk 1 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java (hunk 2 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java]: done

**Comment by xzj7019** [comment]
run buildall

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/PlanUtils.java [hunk 3]
```diff
@@ -125,6 +126,28 @@ public static ImmutableSet<TableIf> getTableSet(LogicalPlan plan) {
...
+    }
+
+    /**
+     * Check if the expression is a column reference.
+     */
+    public static boolean isColumnRef(Expression expr) {
+        return expr instanceof SlotReference
+                && ((SlotReference) expr).getColumn().isPresent()
+                && ((SlotReference) expr).getTable().isPresent();
+    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/PlanUtils.java:149]: use `Expression#isColumnFromTable`

**Review by morrySnow — APPROVED** [review]

**Pull request merged**